uqcsbot/scripts/events.py

```diff
@@ -161,6 +180,43 @@ def get_current_time():
     return datetime.now(tz=BRISBANE_TZ).astimezone(utc)
 
 
+def handle_calendar(calendar) -> List[Event]:
+    """
+    returns a list of events from a calendar
+    """
+    events = []
+    current_time = get_current_time()
+    # subcomponents are how icalendar returns the list of things in the calendar
+    for component in calendar.subcomponents:
+        # we are only interested in ones with the name VEVENT as they
+        # are events
+        if component.name != 'VEVENT':
+            continue
+        elif component.get('RRULE') is not None:
+            # If the until date exists, update it to UTC
+            if component['RRULE'].get('UNTIL') is not None:
+                until = datetime.combine(component['RRULE']['UNTIL'][0], datetime.min.time()) \
+                            .astimezone(utc)
+                component['RRULE']['UNTIL'] = [until]
+            rule = rrulestr('\n'.join([
+                    line for line in component.content_lines()
+                    if line.startswith('RRULE')
+                    or line.startswith('EXDATE')
+                ]), dtstart=component.get('DTSTART').dt)
```

Comment:
This comprehension is a bit of a mess, but I'm not sure how it could be improved.

Reply:
yeah not my best code but im not sure it can be simplified